### ISSUE TITLE
invalid-name False Positive for DataFrame

### ISSUE THREAD

**Issue by nathanjmcdougall** [pylint-dev/pylint#8767]
### Bug description

The pandas `Series` below is not considered constant, whereas `df` is said to be a constant with invalid-name, which I think is a false positive.

```Python
# pylint: disable=missing-module-docstring

import pandas as pd

srs = pd.Series({0: "value"})
srs.name = "name"
df = srs.to_frame().reset_index()
```

### Command used

```shell
pylint bug.py
```

### Pylint output

```shell
************* Module bug
bug.py:13:0: C0103: Constant name "df" doesn't conform to UPPER_CASE naming style (invalid-name)
```

### Expected behavior

I would have expected pylint to run without detecting any issues.

### Pylint version

```shell
pylint 2.17.4
astroid 2.15.5
Python 3.11.1 (tags/v3.11.1:a7a450f, Dec  6 2022, 19:58:39) [MSC v.1934 64 bit (AMD64)]
```

### OS / Environment

Windows 10, Powershell 7.3.4

### Additional dependencies

pandas==2.0.2

**Comment by nathanjmcdougall** [comment]
@Pierre-Sassoulas I don't think this is a duplicate. Consider this similar example:

```
# pylint: disable=missing-module-docstring
# pylint: disable=missing-function-docstring

import pandas as pd


def func(name, column):
    srs = pd.read_csv("data.csv")[column]
    srs.name = name
    df = srs.to_frame().reset_index()
    return df
```
This still gives the same results. This issue seems to only occur specifically when assigning a new name to a `Series`, and then calling `.to_frame()` and `.reset_index()` in succession. It happens regardless of context (function level, module level, etc.), regardless of whether the initial `DataFrame` uses dynamically created objects, etc.

Perhaps I misunderstand #3585, but that issue seems to only be related to module-level constants.

**Comment by nathanjmcdougall** [comment]
@Pierre-Sassoulas I don't think this is a duplicate. Consider this similar example:

```Python
# pylint: disable=missing-module-docstring
# pylint: disable=missing-function-docstring

import pandas as pd


def func(column):
    srs = pd.Series(pd.read_csv("data.csv")[column])
    srs.name = "name"
    my_df = srs.to_frame().reset_index()
    return my_df
```

This still gives the same results. This issue happens regardless of context (function level, module level, etc.), regardless of whether the initial `DataFrame` uses dynamically created objects, etc. It seems to only occur specifically when assigning a new name to a `Series`, and then calling `.to_frame()` and `.reset_index()` in succession, I have tried a number of other combinations of operations but this specific sequence seems to cause issues.

Perhaps I misunderstand #3585, but that issue seems to only be related to module-level constants.

**Comment by nathanjmcdougall** [comment]
My apologies, while experimenting I must have forgotten to save the file. I also can't reproduce it once it's in a function.

Something strange still seems to be going on in examples like this (and again, apologies for my poor care with the last one). In fact, it seems there are other issues too, not just invalid-name.

```Python
# pylint: disable=missing-module-docstring
# pylint: disable=missing-function-docstring

import pandas as pd


def get_data(column):
    return pd.Series(pd.read_csv("data.csv")[column])


srs = get_data("column")
srs.name = "name"
df = srs.to_frame().reset_index()
print(df["column"])
df["new_column"] = 0
```
### Pylint output

```shell
************* Module bug
bug.py:13:0: C0103: Constant name "df" doesn't conform to UPPER_CASE naming style (invalid-name)
bug.py:14:6: E1136: Value 'df' is unsubscriptable (unsubscriptable-object)
bug.py:15:0: E1137: 'df' does not support item assignment (unsupported-assignment-operation)
```

**Comment by Pierre-Sassoulas** [comment]
Did you search for an issue similar to the new false  positive in the issues list ? I'm on mobile but i'm pretty sure there is one ;)

**Comment by nathanjmcdougall** [comment]
You're right, this is clearly the same issue as #3637. I did initially check for duplicates of the `invalid-name` problem, but I couldn't find any mention of it. Actually it turns out the example in the original #3637 issue now also gives `invalid-name` now, so I will add a comment accordingly since nobody has mentioned it yet.

Thanks for helping me stumble my way through this.
